### PR TITLE
DROOLS-2272: [DMN Editor] Move Editor controls inline

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2272

### CODE REVIEW

**Comment by manstis** [comment]
Incorporates https://issues.jboss.org/browse/DROOLS-2296

Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/183
- https://github.com/kiegroup/kie-wb-common/pull/1422
- https://github.com/kiegroup/drools-wb/pull/809

**Comment by manstis** [comment]
This PR renames ```GridCell``` method ```CellSelectionStrategy getSelectionManager()`` to ```CellSelectionStrategy getSelectionStrategy()``` and ```GridData``` method ```setCell(..)``` to ```setCellValue(...)``` introducing a new method ```setCell(..)``` that sets the actual cell rather than its value.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java [hunk 1]
```diff
@@ -357,23 +361,43 @@ public void setColumnDraggingEnabled(final boolean enabled) {
     @Override
     public Range setCell(final int rowIndex,
                          final int columnIndex,
-                         final GridCellValue<?> value) {
+                         final Supplier<GridCell<?>> cellSupplier) {
+        return doSetCell(rowIndex,
+                         columnIndex,
+                         (p) -> cellSupplier.get());
```
[the remark this answers is not in the document]
Comment (jomarko) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java]: Hmmm probably yes, is this case when we just to want set completely new cell at some position?

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/GridData.java [hunk 2]
```diff
@@ -159,12 +160,23 @@ void moveRowsTo(final int index,
...
+     * @param cellSupplier A supplier of new cell instances
      * @return The Range of rows affected by the operation.
      */
     Range setCell(final int rowIndex,
                   final int columnIndex,
-                  final GridCellValue<?> value);
+                  final Supplier<GridCell<?>> cellSupplier);
+
+    /**
+     * Sets a cell at the specified physical coordinate.
```
[the remark this answers is not in the document]
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/GridData.java]: Updated.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java [hunk 3]
```diff
@@ -357,23 +361,43 @@ public void setColumnDraggingEnabled(final boolean enabled) {
...
-            return new Range(rowIndex);
-        }
 
         final int _columnIndex = columns.get(columnIndex).getIndex();
 
         //If we're not merged just set the value of a single cell
         if (!isMerged) {
             ((BaseGridRow) rows.get(rowIndex)).setCell(_columnIndex,
-                                                       value);
+                                                       cellSupplier.apply(new Pair<>(rowIndex, columnIndex)));
```
[the remark this answers is not in the document]
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java]: Changed.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java [hunk 4]
```diff
@@ -389,7 +413,7 @@ public Range setCell(final int rowIndex,
         for (int i = minRowIndex; i <= maxRowIndex; i++) {
             final GridRow row = rows.get(i);
             ((BaseGridRow) row).setCell(_columnIndex,
-                                        value);
+                                        cellSupplier.apply(new Pair<>(i, columnIndex)));
```
[the remark this answers is not in the document]
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java]: Changed.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java [hunk 5]
```diff
@@ -357,23 +361,43 @@ public void setColumnDraggingEnabled(final boolean enabled) {
...
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Range setCellValue(final int rowIndex,
+                              final int columnIndex,
+                              final GridCellValue<?> value) {
+        return doSetCell(rowIndex,
+                         columnIndex,
+                         (p) -> {
```
[the remark this answers is not in the document]
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java]: Updated.

File: uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (manstis) [reply uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/model/impl/BaseGridData.java]: Yes, when using ```setCell(...)``` the existing "cell" is replaced entirely so the ```pair``` has no use. When using ```setCellValue(...)``` the ```pair``` is used to retrieve the existing "cell" and only update its value. In the DMN Editor a use a specific type of ```GridCell``` that has an "in place editor" associated with it. When rows are dragged we update the cell value (row number) and it led to the cell becoming a regular ```BaseGridCell``` and the "in place editor" information lost. Now when a cell's value is changed the existing cell instance is maintained and only the value updated,

**Pull request merged**